src/crawlers/statistics.js

```diff
@@ -44,21 +56,50 @@ export default class Statistics {
         checkParamOrThrow(logIntervalSecs, 'options.logIntervalSecs', 'Number');
         checkParamOrThrow(logMessage, 'options.logMessage', 'String');
 
-
+        this.log = defaultLog.child({ prefix: 'Statistics' });
         this.logIntervalMillis = logIntervalSecs * 1000;
         this.logMessage = logMessage;
+        this.keyValueStore = null;
+        // assign an id while incrementing so it can be saved/restored from KV
+        this.id = Statistics.id++;
+        this.persistStateKey = `STATISTICS_${this.id}_STATE`;
+
+        // initialize by "resetting"
+        this.reset();
+    }
+
+    /**
+     * Set the current statistic instance to pristine values
+     */
+    reset() {
+        if (!this.jobRetryHistogram) {
+            this.jobRetryHistogram = [];
+        } else {
+            this.jobRetryHistogram.length = 0;
+        }
```

Comment:
Why not `this.jobRetryHistogram = []` for both cases?

Reply:
just not too fond of reassigning things that have references elsewhere 😅

Reply:
Oh, got it now. Well, we don't typically prefix internal properties with `_` and just consider all props as internal, unless specifically mentioned. Now I see you made the `jobRetryHistogram` public so clearing the array and the Map below makes sense. On the other hand, I don't think either of them need to be public. Do you think that `getCurrent` is not sufficient public API?